Pivx wallet  pivx: 3.2.0 -> 3.4.0

wucke13 opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

wucke13 committed:
pivx: 3.2.0 -> 3.2.2

wucke13 committed:
altcoins.pivx: 3.2.2 -> 3.3.0

Furthermore, this fixes 2542928

mmahut commented:
This is marked as broken on master, please see 254292858c922fa3bf8c65c75f0868e38c5fe737. If you update supports this, please rebase and drop the broken flag.

wucke13 commented:
Interesting. I was able to build and run the application from this patched derivation. I will look into this soon.

wucke13 commented:
I fixed this by manually speciying `openssl_1_0_2` as dep. Why didn't you do that in the first place in favor of marking the package as broken? @globin @mmahut
At least building is no possible, on master with the most recent version of pivx.

globin commented:
I don't trust upstream who haven't updated from an openssl version that has been EOL for years and includes a number of security issues. That is quite an alarming sign of negligence on their part.

mmahut commented:
@globin has a good point, that the trust in the product is questionable in case they are using EOL libraries, even more if it's a cryptocurrency, however I agree with @wucke13 that we should leave it up to the maintainer to fix any problems with the package and only when we fail to get in touch with the maintainer we should mark the package as broken.

wucke13 commented:
Thank you for the fast responses! Indeed, it depends on `openssl-1.0.x`, neither `openssl-1.1.x` nor any `libressl` will work. While there was some discussion about getting totally rid of `openssl` as a dependency, I wasn't able to quickly find proof of someone actually working on it, so `openssl-1.0.2` is going to stay a dep for some more time, most likely. 
From that standpoint, I'm considering this PR ready for merge, as it won't become any better in the near future. Do you have any other feedback left?

PS: I ~guess~ hope that `pivx` mostly relies on openssl code for their consensus algorithm, and not for actual SSL tunneling. In that case using an outdated version might not be such a big problem?

mmahut commented:
Hard to say how they are using it, but I would maybe open a github issue with them with the request to update, at least in the future? One last rebuild before merging.

@GrahamcOfBorg build altcoints.pivx

wucke13 commented:
@mmahut You had a typo, on `t` too much :)

After doing a quick research, I have to oppose to @globin 's claim of 

> [an] openssl version that has been EOL for years

According to https://mta.openssl.org/pipermail/openssl-announce/2019-May/000151.html openssl-1.0.2 is still maintained, with having its most recent release just four months back in the past. To me this looks like a 'not so well though-through' basis to convince any pivx maintainer of this being an actual issue.

mmahut commented:
@GrahamcOfBorg build altcoins.pivx

wucke13 commented:
Where do they claim that?
https://github.com/PIVX-Project/PIVX/blob/master/doc/build-unix.md only mentiones `openssl-1.0`
Anyway, why didn't the build succeed? Shall I rebase on master once more?

mmahut commented:
Looks like they have stopped using that check yesteday as per https://github.com/PIVX-Project/PIVX/pull/991.

Please rebase as there is a conflict, given the file has been moved from altcoins to blockchains.

wucke13 committed:
pivx: 3.3.0 -> 3.4.0

Added `test_pivx` as test ran by nixpkgs

wucke13 commented:
Ok. The package is still not evaluating with `openssl`, so I'm staying with `openssl_1_0_2` so far. I rebased on master and added a commit for pivx 3.4.0, hopefully this will work. I will edit this post, once compilation succeeded.

mmahut commented:
@GrahamcOfBorg build pivx

wucke13 commented:
@mmahut 
I'm done I guesss. I added a test, added `wrapQtAppsHook` to solve the `qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""` issue. Should build both `pivx` and `pivxd` just fine.

wucke13 commented:
@GrahamcOfBorg build pivx
@GrahamcOfBorg build pivxd

Lassulus commented:
tested with nix-review, started pivx-qt and clicked around a little bit, everything looked fine

Lassulus commented on pkgs/applications/blockchains/pivx.nix:
```diff
@@ -50,12 +56,9 @@ stdenv.mkDerivation rec {
-    # officialy supported version. Either use 1.0.1 or run again configure
-    # with the given option."
-    broken = true;
+    # openssl_1_0_2 should be replaced with opennsl ASAP
```
```suggestion
    # openssl_1_0_2 should be replaced with openssl ASAP
```

wucke13 committed:
Update pkgs/applications/blockchains/pivx.nix

Thanks to @lassulus for fixing this typo!

Co-Authored-By: Lassulus <github@lassul.us>

mmahut commented:
@GrahamcOfBorg eval

wucke13 commented:
@Lassulus Thank you! Can we backport this to `19.09` too?

Lassulus commented:
Sure, can you open a backport PR?
